src/app/results/results.component.ts

```diff
@@ -124,6 +124,7 @@ export class ResultsComponent implements OnInit {
 
       this.presentPage = query['start'] / this.searchdata.rows;
       this.searchdata.query = query['query'];
+      this.store.dispatch(new queryactions.QueryAction(query['query']));
       this.querylook = Object.assign({}, query);
```

Comment:
@nikhilrayaprolu Good work! I was rather going to follow other idea using `localStorage` :smile: